napari/utils/_logging.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+import logging
+from collections import deque
+from typing import TYPE_CHECKING
+
+from psygnal import Signal
+
+_LOG_SEPARATOR = '<NAPARI_LOG_SEPARATOR>'
+
+
+if TYPE_CHECKING:
+    from typing import Any
+
+
+class _LogStream:
+    changed = Signal()
+
+    def __init__(self) -> None:
+        self.logs: deque[tuple[Any, ...]] = deque(maxlen=100_000)
+
+    def write(self, log_msg: str) -> None:
+        logger, level_name, time, thread, msg = log_msg.split(_LOG_SEPARATOR)
+        levels = logging.getLevelNamesMapping()
+        level_value = levels[level_name]
+        self.logs.append((logger, level_value, level_name, time, thread, msg))
+        # TODO: actually save log to a file somewhere so it can be retrieved?
+        self.changed()
+
+    def flush(self) -> None:
+        pass
```

Comment:
What's the dealio here 😂 Maybe add a docstring?

Reply:
Explained in the class docstring (this is required by the superclass).